test/SourceKit/Indexing/index_effective_access_level.swift.response

```diff
@@ -348,202 +362,2198 @@
+              key.line: 84,
+              key.column: 17,
+              key.is_implicit: 1,
+              key.attributes: [
+                {
+                  key.attribute: source.decl.attribute.private
+                }
+              ],
+              key.effective_access: source.decl.effective_access.less_than_fileprivate
+            },
+            {
+              key.kind: source.lang.swift.decl.function.accessor.setter,
+              key.usr: "s:28index_effective_access_level17ScopeKeeperStructV15privateProperty33_2295DDF1454D6A6D9229E8222CD85214LLSivs",
+              key.line: 84,
+              key.column: 17,
+              key.is_implicit: 1,
+              key.attributes: [
+                {
+                  key.attribute: source.decl.attribute.private
+                }
+              ],
+              key.effective_access: source.decl.effective_access.less_than_fileprivate
+            }
+          ],
+          key.attributes: [
+            {
+              key.attribute: source.decl.attribute.private
+            }
+          ],
+          key.effective_access: source.decl.effective_access.less_than_fileprivate
+        },
+        {
+          key.kind: source.lang.swift.ref.struct,
+          key.name: "Int",
+          key.usr: "s:Si",
+          key.line: 84,
+          key.column: 34
+        },
+        {
+          key.kind: source.lang.swift.decl.var.instance,
+          key.name: "publicPropertyWithPrivateSetter",
+          key.usr: "s:28index_effective_access_level17ScopeKeeperStructV31publicPropertyWithPrivateSetterSivp",
+          key.line: 85,
+          key.column: 29,
+          key.entities: [
+            {
+              key.kind: source.lang.swift.decl.function.accessor.getter,
+              key.usr: "s:28index_effective_access_level17ScopeKeeperStructV31publicPropertyWithPrivateSetterSivg",
+              key.line: 85,
+              key.column: 29,
+              key.is_implicit: 1,
+              key.attributes: [
+                {
+                  key.attribute: source.decl.attribute.setter_access.private
+                },
+                {
+                  key.attribute: source.decl.attribute.public
+                }
+              ],
+              key.effective_access: source.decl.effective_access.public
+            },
+            {
+              key.kind: source.lang.swift.decl.function.accessor.setter,
+              key.usr: "s:28index_effective_access_level17ScopeKeeperStructV31publicPropertyWithPrivateSetterSivs",
+              key.line: 85,
+              key.column: 29,
+              key.is_implicit: 1,
+              key.attributes: [
+                {
+                  key.attribute: source.decl.attribute.setter_access.private
+                },
+                {
+                  key.attribute: source.decl.attribute.public
+                }
+              ],
+              key.effective_access: source.decl.effective_access.public
```

Comment:
...Thanks for writing a more thorough set of tests. This is helpful in revealing problems such as this: the "effective access" of the setter shouldn't be 'public' if it's `private(set)`, can it?

Reply:
That's an issue with `getFormalAccess()`, isn't it? For the purposes of this feature, I really only care about the `.decl.`'s access (the getter, but in the "main output"). Do you agree in this case with leaving the explicit getters and setters outputs out of this feature?

Reply:
Sure, that seems fine.

Reply:
There's a separate getSetterFormalAccessScope() for this on AbstractStorageDecl. I think that should be exposed a separate 'effective_setter_access' field on the decl itself (rather than accessors), but can definitely be left as future work for when someone wants that info.